src/main/kotlin/org/rust/lang/core/types/RustTypificationEngine.kt

```diff
@@ -237,3 +240,70 @@ private fun RsImplItem.remapTypeParameters(
                 null
             }
         }.toMap()
+
+private fun inferStructTypeParameters(o: RsStructExpr, item: RsStructItem): RustType {
+    val baseType = item.type
+    if ((baseType as? RustStructOrEnumTypeBase)?.typeArguments.isNullOrEmpty())
+        return baseType
```

Comment:
Let's add braces to this and similar ifs, to be consistent with other code? 

```Kotlin
if ((baseType as? RustStructOrEnumTypeBase)?.typeArguments.isNullOrEmpty()) {
    return baseType
}
```

If you want to avoid braces, and both the condition and the expression are short, you can put the return on the same line:

```Kotlin
if (condition) return foo
```